libyul/Object.cpp

```diff
@@ -65,10 +65,12 @@ string Object::toString(Dialect const* _dialect) const
 
 set<YulString> Object::qualifiedDataNames() const
 {
-	set<YulString> qualifiedNames = name.empty() ? set<YulString>{} : set<YulString>{name};
+	set<YulString> qualifiedNames = name.empty() || contains(name.str(), '.') ? set<YulString>{} : set<YulString>{name};
```

Comment:
I think this is useful for readability, but I know there were arguments against "unneeded" parenthesis:
```suggestion
	set<YulString> qualifiedNames = (name.empty() || contains(name.str(), '.')) ? set<YulString>{} : set<YulString>{name};
```